src/ccall.cpp

```diff
@@ -1690,6 +1691,21 @@ static jl_cgval_t emit_ccall(jl_codectx_t &ctx, jl_value_t **args, size_t nargs)
         ai.decorateInst(tid);
         return mark_or_box_ccall_result(ctx, tid, retboxed, rt, unionall, static_rt);
     }
+    else if (is_libjulia_func(jl_get_tls_world_age)) {
+        ++CCALL_STAT(jl_get_tls_world_age);
+        assert(lrt == ctx.types().T_size);
+        assert(!isVa && !llvmcall && nccallargs == 0);
+        JL_GC_POP();
+        Value *task = get_current_task(ctx);
+        const int world_age_offset = offsetof(jl_task_t, world_age);
+        Value *p_world_age = ctx.builder.CreateInBoundsGEP(ctx.types().T_size, task, ConstantInt::get(ctx.types().T_size, world_age_offset / ctx.types().sizeof_ptr));
+        setName(ctx.emission_context, p_world_age, "task_world_age_ptr");
+        LoadInst *world_age = ctx.builder.CreateAlignedLoad(ctx.types().T_size, p_world_age, Align(sizeof(ctx.types().sizeof_ptr)));
+        setName(ctx.emission_context, world_age, "task_world_age");
+        jl_aliasinfo_t ai = jl_aliasinfo_t::fromTBAA(ctx, ctx.tbaa().tbaa_const);
```

Comment:
Is this annotation correct?

I'm pretty sure from LLVM's point-of-view, functions can manipulate the world age.

Reply:
Hmm, I wonder if we have a TBAA annotation that is proper for this, because I don't want llvm thinking this alias anything else.

Reply:
@vchuravy or @vtjnash any ideas?

Reply:
I think the semantics of world age are under specified / not documented at the moment.

Most places that change the world age will reset it afterwards. E.g. `invokelatest`.

Do I do think we could define and document that world_age must be constant within a function

Reply:
> I do think we could define and document that world_age must be constant within a function

I'm assuming you mean in the Julia-side semantics - We do have cases in generated LLVM code that we temporarily manipulate the world age (in a way not visible to `Base.tls_world_age()` etc.)

I think `!invariant.load` is too strict to accommodate that, but `!invariant.group` looks like it could do the trick. That should let us annotate that these specific loads are constant, even though the particular memory location is not constant for all loads/stores.

Reply:
invariant.group or another option is to always move this load to the function header like we do for other tls things